### PR TITLE
ci: allow users to override env vars

### PR DESCRIPTION
**Pull request by skoh7645**
### Checklist:

- [x] Read the [Code of Conduct](https://github.com/appsody/website/blob/master/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/appsody/website/blob/master/CONTRIBUTING.md).

- [x] Followed the [commit message guidelines](https://github.com/appsody/website/blob/master/CONTRIBUTING.md#commit-message-guidelines).

- [x] Stack adheres to [Appsody stack structure](https://github.com/appsody/website/blob/master/content/docs/stacks/stack-structure.md).

### Modifying an existing stack:
Added ability for users to set environment variables before running appsody stacks build scripts.

### Related Issues:
Fixes: https://github.com/appsody/appsody/issues/166

### CODE REVIEW

File: ci/build.sh [hunk 1]
```diff
@@ -9,6 +9,27 @@ fi
...
+    export DOCKERHUB_ORG=appsody
+fi
+
+# url for downloading released assets
+if [ -z $RELEASE_URL ]
+then
+    RELEASE_URL="https://github.com/$TRAVIS_REPO_SLUG/releases/download"
+fi
+
+if [ -z $STACK_LIST ]
```
Comment (mhamwala) [inline ci/build.sh]: this should be `$STACKS_LIST`

Comment (skoh7645) [reply ci/build.sh]: thank you! fixed now

File: ci/build.sh [hunk 2]
```diff
@@ -9,6 +9,27 @@ fi
 
 base_dir="$(cd "$1" && pwd)"
 
-. $base_dir/ci/list.sh $base_dir
+if [ -z $REPO_LIST ]
+then
+    REPO_LIST="experimental incubator stable"
```
Comment (mhamwala) [inline ci/build.sh]: add export before REPO_LIST

File: ci/build.sh [hunk 3]
```diff
@@ -9,6 +9,27 @@ fi
...
+# dockerhub org for publishing stack
+if [ -z $DOCKERHUB_ORG ]
+then
+    export DOCKERHUB_ORG=appsody
+fi
+
+# url for downloading released assets
+if [ -z $RELEASE_URL ]
+then
+    RELEASE_URL="https://github.com/$TRAVIS_REPO_SLUG/releases/download"
```
Comment (mhamwala) [inline ci/build.sh]: add export before RELEASE_URL

File: ci/package.sh [hunk 4]
```diff
@@ -6,26 +6,24 @@ if [ -z "$1" ]
 then
     echo "One argument is required and must be the base directory of the repository."
     exit 1
+elif [ -z $DOCKERHUB_ORG] || [ -z $RELEASE_URL ]
```
Comment (mhamwala) [inline ci/package.sh]: add a space after `DOCKERHUB_ORG`

File: ci/build.sh (hunk 2 above)
Comment (skoh7645) [reply ci/build.sh]: fixed

File: ci/build.sh (hunk 3 above)
Comment (skoh7645) [reply ci/build.sh]: thank you

File: ci/package.sh (hunk 4 above)
Comment (skoh7645) [reply ci/package.sh]: thanks

File: ci/list.sh [hunk 5]
```diff
@@ -26,16 +26,19 @@ then
...
     done
 else
+
     if [ $TRAVIS_TAG ]
     then
         stack_id=`echo ${TRAVIS_TAG/-v[0-9]*/}`
         echo "Listing stacks for this release"
+    elif [ -z $STACKS_LIST ]
+    then
+        echo "You need to set a value for the following environment variables:"
```
Comment (neeraj-laad) [inline ci/list.sh]: I do not think we need this. `list.sh` will set this variable if it is not set

Comment (skoh7645) [reply ci/list.sh]: Thank you, I have fixed this in the most recent commit

File: ci/package.sh [hunk 6]
```diff
@@ -6,26 +6,26 @@ if [ -z "$1" ]
 then
     echo "One argument is required and must be the base directory of the repository."
     exit 1
+elif [ -z $DOCKERHUB_ORG ] || [ -z $RELEASE_URL ]
```
Comment (neeraj-laad) [inline ci/package.sh]: we can delete this check too is we are setting this in env.sh?

File: ci/env.sh [hunk 7]
```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+set -e
+
+# values can also be set for env vars DOCKER_PASSWORD, DOCKER_USERNAME and DOCKER_REGISTRY
```
Comment (neeraj-laad) [inline ci/env.sh]: You can set the following environment variables to specify your docker credentials or docker registry

File: ci/env.sh [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+
```
Comment (neeraj-laad) [inline ci/env.sh]: this is the default list of repos that we need to build index for

**Pull request merged**